### DISCUSSION TITLE
Not able to use nodejs fs module

### DISCUSSION THREAD

**Discussion by saalimzafar** [vercel/next.js#41651] [Help]
import fs from [node:fs | fs from fs | from node]
---------
wait  - compiling /file...
#1 error - node:fs
Module build failed: UnhandledSchemeError: Reading from "node:fs" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
Could not find files for /file in .next/build-manifest.json
Could not find files for /file in .next/build-manifest.json
wait  - compiling...
event - compiled successfully in 52.1s (179 modules)
wait  - compiling /_error...
event - compiled successfully in 753 ms (179 modules)
wait  - compiling /file...
#2 error - node:fs
Module build failed: UnhandledSchemeError: Reading from "node:fs" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
Could not find files for /file in .next/build-manifest.json
Could not find files for /file in .next/build-manifest.json
wait  - compiling...
event - compiled successfully in 34.6s (179 modules)
wait  - compiling /file...
#3 error - ./pages/file.js:1:0
Module not found: Can't resolve 'node'
> 1 | import fs from 'node'
  2 | export default function File() {
  3 | return (
  4 | <>File Database System</>

https://nextjs.org/docs/messages/module-not-found
wait  - compiling /_error...
error - ./pages/file.js:1:0
Module not found: Can't resolve 'node'
> 1 | import fs from 'node'
  2 | export default function File() {
  3 | return (
  4 | <>File Database System</>

https://nextjs.org/docs/messages/module-not-found
wait  - compiling...
error - ./pages/file.js:1:0
Module not found: Can't resolve 'fs'
> 1 | import fs from 'fs'
  2 | export default function File() {
  3 | return (
  4 | <>File Database System</>

https://nextjs.org/docs/messages/module-not-found
wait  - compiling...
event - compiled successfully in 47.3s (179 modules)
wait  - compiling /_error...
event - compiled successfully in 819 ms (179 modules)
wait  - compiling /file...
error - ./pages/file.js:1:0
Module not found: Can't resolve 'fs'
> 1 | import writeFile from 'fs'
  2 | export default function File() {
  3 | return (
  4 | <>File Database System</>

https://nextjs.org/docs/messages/module-not-found
Could not find files for /file in .next/build-manifest.json

------

**Comment by M1ck0** [comment]
Just add this to `next.config.js` and you'll be good to go

```js
webpack: (config) => {
    config.resolve.fallback = {
      fs: false,
      net: false,
      dns: false,
      child_process: false,
      tls: false,
    };

    return config;
  },
```

**Reply by saalimzafar** [reply]
Thanks for your response, but still it's not working.
I'm not getting the errors after this, but empty object when I try to find the type in alert.

**Reply by M1ck0** [reply]
That's strange. This fixed issue for me.
